conbench/entities/benchmark_result.py

```diff
@@ -1159,7 +1159,10 @@ class _BenchmarkResultCreateSchema(marshmallow.Schema):
             )
         },
     )
-    batch_id = marshmallow.fields.String(required=True)
+    batch_id = marshmallow.fields.String(
+        # this lacks specification and should probably not be required
+        required=True
+    )
```

Comment:
This is user-customizable, but in almost all cases this is the interaction of benchmark name and run ID, i.e. the collection of cases for a benchmark on a run.

I see value in that idea (e.g. when one case is weird, look at the others), but we can get the same result by grouping by name and run ID, which could be the default if it's not set.

Reply:
Yes. We've had this conversation, and noted down our findings :) https://github.com/conbench/conbench/issues/880.

My code comment there was once again JP trying to 'quickly do something' while I went over this.

Reply:
https://github.com/conbench/conbench/pull/1427/files#diff-9d75d1bf6ac23bfc056a410721cf4c9bccd8b302f8e3e83e4b4bf1698d0eaad7R1164